packages/components/src/utils/element-rect.ts

```diff
@@ -171,6 +174,16 @@ export function useTrackElementOffsetRect(
 		}
 	}, [ setElement, targetElement ] );
 
+	// Escape hatch to force a remeasurement when something else changes rather
+	// than the target elements' ref or size (for example, the target element
+	// can change its position within the tablist).
+	useLayoutEffect( () => {
+		measure();
+		// `measure` is a stable function, so it's safe to omit it from the deps array.
+		// deps can't be statically analyzed by ESLint
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
ESLint disabling is a problem for React Compiler.

I'd prefer if we leave the warning in place. There are plenty of those warnings across the codebase already. Yes, we'll have to deal with them, but that can be separately addressed.

Reply:
As discussed, we will address this at a later point, so that we can work across the whole reporsitory

Reply:
Tackling this in #66324.